util/fetch.go

```diff
@@ -76,8 +77,13 @@ func JWTTokenSource(ctx context.Context, settings *Settings) (oauth2.TokenSource
 	if err != nil {
 		return nil, err
 	}
-	ts, err := google.JWTAccessTokenSourceFromJSON(creds.JSON, settings.Audience)
-	return ts, err
+	if settings.Audience != "" {
+		return google.JWTAccessTokenSourceFromJSON(creds.JSON, settings.Audience)
+	} else if settings.Scope != "" {
+		return google.JWTAccessTokenSourceWithScope(creds.JSON, settings.Scope)
+	} else {
+		return nil, errors.New("neither audience or scope is provided.")
```

Comment:
nit: I think the golang convention is not to add periods "." at the end of error messages. (Don't forget to update the golden file if you decide to remove the ".")